ocs_ci/ocs/platform_nodes.py

```diff
@@ -2420,3 +2420,62 @@ def restart_nodes_by_stop_and_start_teardown(self):
                 cmd = f"ibmcloud ks worker reboot --cluster {cluster_id} --worker {not_ready_node} -f"
                 out = run_cmd(cmd)
                 logger.info(f"Node restart command output: {out}")
+
+    def create_nodes(self, node_conf, node_type, num_nodes):
+        """
+        Create node on IBM Cloud.
+
+        Args:
+            node_conf (dict): of node configuration
+            node_type (str): type of node to be created
+            num_nodes (int): Number of node instances to be created
+
+        Returns:
+           list: of IBMCloudNode objects
+
+        """
+        logger.info("creating new node")
+        login()
+
+        worker_nodes = get_nodes(node_type="worker")
+        provider_id = worker_nodes[0].get()["spec"]["providerID"]
+        cluster_id = provider_id.split("/")[5]
+
+        cmd = f"ibmcloud ks workers --cluster {cluster_id} --output json"
+        out = run_cmd(cmd)
+        worker_nodes = json.loads(out)
+
+        no_of_nodes = len(worker_nodes) + int(num_nodes)
+        logger.info(f"number of nodes going to be add are : {num_nodes}")
+
+        cmd = (
+            f"ibmcloud ks worker-pool resize --cluster {cluster_id} --worker-pool default"
+            f"  --size-per-zone {no_of_nodes}"
+        )
+        run_cmd(cmd)
+
+        logger.info("Waiting for 600 seconds for node to ready")
+        time.sleep(600)
```

Comment:
There should be function which will check if node is ready and return yes if it is ready.  Here you should not wait 10 minutes as it can be sooner probably - then you should use our timeout sampler functionality which will repeat the call of the function which will check if node is ready - if not it's continuously re-trying and once done -it's continue - so let
s say node is ready in 5 minutes - then  you don't have to spend here 10 minutes but only half of the time.

You can see example of timeout samplers e.g. here:
https://github.com/red-hat-storage/ocs-ci/blob/6ecc96ca640e96448a16e2271f5553a7a6248c12/tests/manage/pv_services/test_verify_rwo_using_dc_pod.py#L79

or look for it's usage on the other places.

Reply:
Done